### PR TITLE
Update documentation to update file location links on bastion

### PR DESCRIPTION
**Pull request by pli888**
# Pull request for issue: [#1423](https://github.com/gigascience/gigadb-website/issues/1423) Replace urls of non-peculiar datasets available from wasabi on staging (and capture how long it took)

This is a pull request which provides extra information for using the files metadata console tool to update File location URLs on bastion server.

## Any changes to documentation?

The File Metadata Console's README.md file has been updated in the [Staging and live environments](https://github.com/pli888/gigadb-website/blob/bastion-replace-urls/gigadb/app/tools/files-metadata-console/README.md#staging-and-live-environments) section. There is some new advice for preparing to update File location links before going live on AWS infrastructure. The latest documentation also says that processing production GigaDB data should take up to 1 hour 30 min to update all the file locations for all public datasets.

Boilerplate documentation code at the beginning of the README.md file has been removed as I thought it was distracting from the purpose of the tool.

### CODE REVIEW

**Review by kencho51 — COMMENTED** [review]
Hi @pli888,

**praise:** Good to know where to find the big dataset file.
**praise:**  Good to have have documented how to use the tool practically in staging and live environments.
**thought:** The example cmd will need to be updated if there is a new and huge dataset,  is it better to make it more robust by saying something like,
```
$ docker run --rm "registry.gitlab.com/$GITLAB_PROJECT/production-files-metadata-console:$GIGADB_ENV" ./yii update/urls --separator=/pub/ --stop=102457 --next=50 --exclude='list of DOI to exclude`
For example:
$ docker run --rm "registry.gitlab.com/$GITLAB_PROJECT/production-files-metadata-console:$GIGADB_ENV" ./yii update/urls --separator=/pub/ --stop=102457 --next=50 --exclude='100050,100115,100157,100242,100310,100396,100443,100608,100622,100707,100849,102425,102431,102440,102441'
```

File: gigadb/app/tools/files-metadata-console/README.md [hunk 1]
```diff
@@ -143,7 +143,14 @@ and see file links that point to Wasabi.
...
 because they contain too many files for the files metadata console tool to 
-process. You can identify these datasets using the big datasets spreadsheet as 
+process. These datasets need to be manually updated using SQL commands. We begin
+by starting a database transaction so that each manual dataset update is 
+contained within a unit of work:
+```
+$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:$GIGADB_ENV -c "BEGIN TRANSACTION;"
+```
+
+You can identify these datasets using the big datasets spreadsheet as
```
Comment (kencho51) [inline gigadb/app/tools/files-metadata-console/README.md:153]: Hi @pli888,

**issue:** In order to use the database transaction, all sql cmd need to be executed at once, which is not the case in this doc. If I followed the steps in this doc, all sql work is executed outside the transaction, for example:
```
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "BEGIN TRANSACTION;"
BEGIN
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "Update dataset set ftp_site = REPLACE(ftp_site, 'https://ftp.cngb.org/pub/gigadb/pub/', 'https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/') where identifier = '100050';"
UPDATE 1
[centos@ip-10-99-0-113 ~]$ ocker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select id, identifier, ftp_site from dataset where identifier = '100050';"
-bash: ocker: command not found
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select id, identifier, ftp_site from dataset where identifier = '100050';"
 id | identifier |                                            ftp_site                                            
----+------------+------------------------------------------------------------------------------------------------
 44 | 100050     | https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/
(1 row)

[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "Update file set location = REPLACE(location, 'https://ftp.cngb.org/pub/gigadb/pub/', 'https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/') where dataset_id =44;"
UPDATE 307
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select location from file where dataset_id = '44' limit 10;;"
                                                                              location                                                                              
--------------------------------------------------------------------------------------------------------------------------------------------------------------------
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100210_I354_FC6176HAAXX_L2_WHElgqDAADBBAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091225_I647_FC61B36AAXX_L3_WHElgqDAADWAAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/ReseqRawData/091107_I655_FC61924AAXX_L3_WHElwuRACDICAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100210_I354_FC6176HAAXX_L2_WHElgqDAADBBAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/ReseqRawData/091122_I328_FC61AVHAAXX_L7_WHElwuRACDIBAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091106_I360_FC42T1VAAXX_L6_WHElgqDAADFAAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091122_I647_FC618YNAAXX_L6_WHElgqDAADIAAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091225_I647_FC61B36AAXX_L4_WHElgqDAADLAAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100603_I321_FC704P2AAXX_L6_WHElgqDADDLCAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091029_I77_FC42RTGAAXX_L2_WHElgqDABDMAAPE_1.fq.gz
(10 rows)

[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "COMMIT TRANSACTION;"
COMMIT
WARNING:  there is no transaction in progress
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select location from file where dataset_id = '44' limit 10;"
                                                                              location                                                                              
--------------------------------------------------------------------------------------------------------------------------------------------------------------------
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100603_I321_FC704P2AAXX_L6_WHElgqDADDLCAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091029_I77_FC42RTGAAXX_L2_WHElgqDABDMAAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091105_I361_FC42T1HAAXX_L4_WHElgqDABDFBAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091106_I360_FC42T1VAAXX_L6_WHElgqDAADFAAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100118_I638_FC4343PAAXX_L6_WHElgqDAADLAAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100514_I638_FC61KC4AAXX_L2_WHElgqDADDWEAPE_2.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100210_I653_FC612YKAAXX_L4_WHElgqDABDMAAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091106_I360_FC42T1VAAXX_L5_WHElgqDAADFBAPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/091029_I77_FC42RTGAAXX_L5_WHElgqDAADMABPE_1.fq.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100050/A/GenomicRawReads/100212_I361_FC617AHAAXX_L6_WHElgqDADDTBBPE_1.fq.gz
(10 rows)

[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "COMMIT;"
WARNING:  there is no transaction in progress
COMMIT
```
**suggestion:** You could have a look of the following output from my staging:
```
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select id, identifier, ftp_site from dataset where identifier = '100115';"
 id  | identifier |                             ftp_site                             
-----+------------+------------------------------------------------------------------
 177 | 100115     | https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115
(1 row)
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "BEGIN; Update dataset set ftp_site = REPLACE(ftp_site, 'https://ftp.cngb.org/pub/gigadb/pub/', 'https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/') where identifier = '100115'; Update file set location = REPLACE(location, 'https://ftp.cngb.org/pub/gigadb/pub/', 'https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/') where dataset_id = 177; ROLLBACK;"
ROLLBACK
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select id, identifier, ftp_site from dataset where identifier = '100115';"
 id  | identifier |                             ftp_site                             
-----+------------+------------------------------------------------------------------
 177 | 100115     | https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115
(1 row)

[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select location from file where dataset_id = '177' limit 10;"
                                              location                                              
----------------------------------------------------------------------------------------------------
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/GATK/MDA-2_66.snps.vqsr.vcf.gz
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/GATK/SW480-SCD.indels.vqsr.vcf.gz
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/GATK/SW480-SCD.snps.vqsr.vcf.gz
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/GATK/DOP-1_97.indels.vqsr.vcf.gz
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/BWA/MDA-2_M6.dedup.bam
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/BWA/MDA-2_46.dedup.bam
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/BWA/MDA-2_66.dedup.bam
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/BWA/MDA-2_M16.dedup.bam
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/BWA/MDA-3_45.dedup.bam
 https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100115/GATK/SW480-1.snps.vqsr.vcf.gz
(10 rows)

[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "BEGIN; Update dataset set ftp_site = REPLACE(ftp_site, 'https://ftp.cngb.org/pub/gigadb/pub/', 'https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/') where identifier = '100115'; Update file set location = REPLACE(location, 'https://ftp.cngb.org/pub/gigadb/pub/', 'https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/') where dataset_id = 177; COMMIT;"
COMMIT
[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select id, identifier, ftp_site from dataset where identifier = '100115';"
 id  | identifier |                                           ftp_site                                            
-----+------------+-----------------------------------------------------------------------------------------------
 177 | 100115     | https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115
(1 row)

[centos@ip-10-99-0-113 ~]$ docker run --rm  --env-file ./db-env registry.gitlab.com/$GITLAB_PROJECT/production_pgclient:staging -c "select location from file where dataset_id = '177' limit 10;"
                                                            location                                                            
--------------------------------------------------------------------------------------------------------------------------------
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/BWA/MDA-2_46.dedup.bam
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/BWA/MDA-2_66.dedup.bam
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/BWA/MDA-2_M16.dedup.bam
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/BWA/MDA-3_45.dedup.bam
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/GATK/SW480-1.snps.vqsr.vcf.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/BWA/SW480-SCD.markdup.bam
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/BWA/MDA-2_47.dedup.bam
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/readme.txt
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/GATK/SW480-5.indels.vqsr.vcf.gz
 https://s3.ap-northeast-1.wasabisys.com/gigadb-datasets/live/pub/10.5524/100001_101000/100115/GATK/MDA-2_66.indels.vqsr.vcf.gz
(10 rows)

[centos@ip-10-99-0-113 ~]$ 
```

**Comment by rija** [comment]
Hi @pli888, @kencho51 

Here's how you can do the transaction over multiple commands:

https://dba.stackexchange.com/a/13520/236517

**Comment by rija** [comment]
@pli888, @kencho51 

Thinking more about it, if we make a mistake in the updates, we can always use the DB restoration procedure, so I don't think we should try to wrap our update commands in transaction, even if we now know how to do it. it's easier that way.
We should just make sure to start the manual update at a time when we can do restoration to point in time recovery that will restore to the state before our manual update.

**Review by rija — CHANGES_REQUESTED** [review]
Hi @pli888, @kencho51

**issue**: the manual update of urls is quite inefficient and will create deadlocks for other users if they try to use the database during the operation
**suggestion**: use a temporary table and SQL based batches as described in 
https://minhajuddin.com/2020/10/17/how-to-do-batch-updates-in-postgresql/
**issue**: the automated update of urls using files-metadata-console has same problems as the manual update
**suggestion**: don't use the PHP app, let's do ALL the url replacements using SQL and the approach in previous suggestion and batch over files rows, NOT datasets (so it doesn't matter whether a dataset has 200000 files or 5)
**issue**: there maybe a sentiment that we need to verify each batch's correctness. This is not necessary
**suggestion**: After running the update, we can generate a SQL report to return number of files with still the old location, with the new location and any other values, and if there are discrepancies, we can, run more target SQL queries,  and tweak the location replacement queries, then try again
**issue**: Too many manually triggered jobs are described in the README.md, that takes a lot of times of the operator and is error prone
**suggestion**: Have a single bash script that do it all, we can run it in one go, then we make SQL reports to verify that every bit is done correctly. If issues,  tweak the problematic area, then restart that unique bash script
**note**: We need to beef up the size of the DB server, and it will help to some extent (and beyond this use casetoo),  but it won't solve the deadlock problem or the operator manual work, so the above suggestions are still needed even after scaling up the server

**Comment by pli888** [comment]
>suggestion: Have a single bash script that do it all, we can run it in one go, then we make SQL reports to verify that every bit is done correctly. If issues, tweak the problematic area, then restart that unique bash script

There is now a bash script in `scripts/updateUrls.sh` which can update dataset ftp_site and file location URLs into Wasabi links. Documentation to test this script can be found in `scripts/README.md`. There are comments in `scripts/updateUrls.sh` which explains how URLs are updated but it basically involves copying the links in the dataset and file tables into temporary database tables. The URLs in the temporary tables are then updated into Wasabi links which are copied over into the dataset and file tables. A new task in bastion playbook copies the `updateUrls.sh` script into the bastion server in order for it to be used to transform URLs on staging and live servers.

File: gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  file
+WHERE
+  location LIKE '%' || :'ftp_site_parrot' || '%'
+OR
+  location LIKE '%' || :'ftp_site_climb' || '%'
+OR
+  location LIKE '%' || :'ftp_site_cngb' || '%'
+ORDER BY
+  dataset_id ASC;
+\echo Created dataset_changes temporary table
```
Comment (kencho51) [inline gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh]: **typo:** should be `file_changes` ?

File: gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  (
+    SELECT
+      id,
+      location
+    FROM
+      file_changes
+  ) AS subquery
+WHERE
+  file.id=subquery.id;
+\echo Copied URLs from temporary table into dataset table ftp_site column
```
Comment (kencho51) [inline gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh]: **typo:** should be `file table location column`?

**Review by kencho51 — COMMENTED** [review]
Hi @pli888,

**praise:** The `updateUrls.sh` works as expected to replace the existing ftp url with the wasabi one in my dev database and also staging database.
**praise:** It is good idea to `--set AUTOCOMMIT=off` explicitly in the script, which helps to prevent error.
**typo:** Please refer to the inline comments.
**question:** Why creating a new README file in the script dir, but not appending its content to gigadb/app/tools/files-metadata-console/README.md?  In `gigadb/app/tools/files-metadata-console/README.md`, there are already section about how to update the ftp urls.
**chore:** Resolve the `CHANGELOG.md` conflicts .

**Review by rija — APPROVED** [review]
Hi @pli888,

**praise**: the url updater shell script works on my dev environment, and on my staging environment
**praise**: nice technique for passing variables to the SQL closures.
**praise**: there is a unit test written
**issue** (non-blocking): the unit test can only be run once
**suggestion**: add a `teardown()` block to the test that reset the database so the test can be run any number of times without manually starting/stopping the container services:
```
teardown () {  
      cd ../../../../  
      ./ops/scripts/setup_devdb.sh dev  
      cd gigadb/app/tools/files-metadata-console  
    }
```
**praise**: nice SQL technique (temporary table) for efficiently and with reduced risk of blocking updating urls
**nitpick**: regarding the use of `vars.bucketdir`, [PostgreSQL documentation](https://www.postgresql.org/docs/14/runtime-config-custom.html) indicates the left part of the dot, is namespace for extensions. I can see two situations were having chosen that namespace could cause us problems:
  * Someone genuinely creates a `vars` extension and AWS adds it to the list of extensions installed by default in RDS, and that extension doesn't have a parameter `bucketdir`, this will cause server exception
  * a threat actor who knows that everyone who have checked Stack Overflow (or Chat GPT)  for a solution to setting up a custom session parameter as a variable  will use the `vars` namespace, may create a malware of the same name and may try to get it installed on RDS instances using exploits or whatnot
  
**suggestion**: use a namespace specific to us (our organisation, project)
**praise**: the tool has a clear README.md
**chore**: rebase as CHANGELOG is in conflict

Provided the branch will be rebased, I'm happy to approve this PR.

File: gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh (hunk 2 above)
Comment (pli888) [reply gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh]: Corrected now.

File: gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh (hunk 3 above)
Comment (pli888) [reply gigadb/app/tools/files-metadata-console/scripts/updateUrls.sh]: Corrected now.

**Comment by pli888** [comment]
@kencho51 

>typo: Please refer to the inline comments.

Fixed now in https://github.com/gigascience/gigadb-website/pull/1448/commits/444716879e24f34f871a788a481d9e90edaa8bb1.

>question: Why creating a new README file in the script dir, but not appending its content to gigadb/app/tools/files-metadata-console/README.md? In gigadb/app/tools/files-metadata-console/README.md, there are already section about how to update the ftp urls.

Because the files-metadata-console/scripts/README.md file contains documentation for the `updateUrls.sh` shell script, whilst files-metadata-console/README.md contains documentation for the Yii2 console application. They have similar functionality but they are separate applications, and I think it's better to have shorter README files if possible.

> chore: Resolve the CHANGELOG.md conflicts

Done.

**Review by kencho51 — APPROVED** [review]
Hi @pli888,

LGTM, happy to approve now.

**Comment by pli888** [comment]
@rija 

> suggestion: add a teardown() block to the test that reset the database so the test can be run any number of times without manually starting/stopping the container services

Done in https://github.com/gigascience/gigadb-website/pull/1448/commits/1208308b656df2539e414a81379f6fb2803becaa

> suggestion: use a namespace specific to us (our organisation, project)

Done in https://github.com/gigascience/gigadb-website/pull/1448/commits/fe198f0f6f4034fcf65630a28141f5345312e01a

**Pull request merged**